File: Solutions/CF344A.py
number_of_magnets = int(input())

if number_of_magnets == 0:
    print(0)

else:
    number_of_groups = 0

    prev_magnet = input()
    number_of_magnets -= 1
    number_of_groups += 1

    while number_of_magnets:
        new_magnet = input()
        if new_magnet != prev_magnet:
            prev_magnet = new_magnet
            number_of_groups += 1

        number_of_magnets -= 1

    print(number_of_groups)

File: Solutions/CF344A.cpp
// Magnets
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <unordered_map>
#include <math.h>
#include <algorithm>
#include <iomanip>
using namespace std;

#define lli long long int

void magnets(lli number_of_magnets)
{
    lli prev_magnet;
    cin >> prev_magnet;
    lli number_of_groups = 1;

    while (number_of_magnets)
    {
        lli new_magnet;
        cin >> new_magnet;

        if (new_magnet != prev_magnet)
        {
            prev_magnet = new_magnet;
            number_of_groups++;
        }

        number_of_magnets--;
    }

    cout << number_of_groups << endl;
}

int main(int argc, char const *argv[])
{
    lli number_of_magnets;
    cin >> number_of_magnets;

    if (number_of_magnets == 0)
        cout << 0 << endl;
    else
        magnets(number_of_magnets - 1);

    return 0;
}
